docroot/sites/classrooms.uiowa.edu/modules/classrooms_core/classrooms_core.module

```diff
@@ -546,20 +576,65 @@ function classrooms_core_theme($existing, $type, $theme, $path) {
  */
 function classrooms_core_form_views_exposed_form_alter(&$form, FormStateInterface $form_state, $form_id) {
   $view = $form_state->get('view');
-  if ($view->id() === 'room_list') {
-    $building_filter = 'field_room_building_id_target_id';
-    if (isset($form[$building_filter])) {
-      uiowa_core_config_entity_filter_select($form[$building_filter], 'building');
-
-      // Change the label from default `- Any -`.
-      $form[$building_filter]['#empty_option'] = t('Building');
-
-      // Hack to further treat this field as a select as it relates to BEF.
-      // Assumes `field_room_type_target_id` is present
-      // on the form and is a select.
-      $form[$building_filter]['#process'] = $form['field_room_type_target_id']['#process'];
-      $form[$building_filter]['#pre_render'] = $form['field_room_type_target_id']['#pre_render'];
-    }
+  $view_display = $view->getDisplay();
+  $block_id = $view_display->display['id'];
+
+  switch ($block_id) {
+    case 'block_rooms':
+      $building_filter = 'field_room_building_id_target_id';
+      if (isset($form[$building_filter])) {
+        uiowa_core_config_entity_filter_select($form[$building_filter], 'building');
+
+        // Change the label from default `- Any -`.
+        $form[$building_filter]['#empty_option'] = t('Building');
+
+        // Hack to further treat this field as a select as it relates to BEF.
+        // Assumes `field_room_type_target_id` is present
+        // on the form and is a select.
+        $form[$building_filter]['#process'] = $form['field_room_type_target_id']['#process'];
+        $form[$building_filter]['#pre_render'] = $form['field_room_type_target_id']['#pre_render'];
+      }
+      break;
+
+    case 'block_programmed_rooms_list':
+      $building_filter = 'field_room_building_id_target_id';
+      if (isset($form[$building_filter])) {
+        uiowa_core_config_entity_filter_select($form[$building_filter], 'building');
+
+        // Change the label from default `- Any -`.
+        $form[$building_filter]['#empty_option'] = t('Building');
+
+        // Hack to further treat this field as a select as it relates to BEF.
+        $form[$building_filter]['#process'] = [
+          [
+            'Drupal\Core\Render\Element\Select',
+            'processSelect',
+          ],
+          [
+            'Drupal\Core\Render\Element\Select',
+            'processAjaxForm',
+          ],
+          [
+            'Drupal\inline_form_errors\RenderElementHelper',
+            'processElement',
+          ],
+          [
+            '\Drupal\Core\Render\Element\RenderElement',
+            'processGroup',
+          ],
+        ];
+        $form[$building_filter]['#pre_render'] = [
+          [
+            'Drupal\Core\Render\Element\Select',
+            'preRenderSelect',
+          ],
+          [
+            '\Drupal\Core\Render\Element\RenderElement',
+            'preRenderGroup',
+          ],
+        ];
+      }
+      break;
   }
```

Comment:
Do we need to keep these as separate cases? They seem to end up with the same result in both cases, but in the first we're using an existing field to build it, and the second we explicitly define it.

Instead of a switch, maybe an `if in_array(...)` still to cover our bases?

Reply:
```suggestion
  if (in_array($block_id, ['block_rooms', 'block_programmed_rooms_list'])) {
    $building_filter = 'field_room_building_id_target_id';
    if (isset($form[$building_filter])) {
      uiowa_core_config_entity_filter_select($form[$building_filter], 'building');

      // Change the label from default `- Any -`.
      $form[$building_filter]['#empty_option'] = t('Building');

      // Hack to further treat this field as a select as it relates to BEF.
      $form[$building_filter]['#process'] = [
        [
          'Drupal\Core\Render\Element\Select',
          'processSelect',
        ],
        [
          'Drupal\Core\Render\Element\Select',
          'processAjaxForm',
        ],
        [
          'Drupal\inline_form_errors\RenderElementHelper',
          'processElement',
        ],
        [
          '\Drupal\Core\Render\Element\RenderElement',
          'processGroup',
        ],
      ];
      $form[$building_filter]['#pre_render'] = [
        [
          'Drupal\Core\Render\Element\Select',
          'preRenderSelect',
        ],
        [
          '\Drupal\Core\Render\Element\RenderElement',
          'preRenderGroup',
        ],
      ];
    }
  }
```

Reply:
Good question @cory-skeers, @e-marie-w or @joewhitsitt did you work on this part?

Reply:
Yeah that will work. Originally, we were able to target the existing fields to use for block_rooms, but another select was not available on block_programmed_rooms_list to use. The warning at the time is that if BEF or something changes how these are processed, we may need to update them.

Reply:
Ah, that makes sense.

> if BEF or something changes how these are processed, we may need to update them.

Think it's worth leaving part of the field copying in place for the case where it's there? If a BEF change would break it then it would only break one and we'd have an easy example on how to fix it.

Something like 
```
if (isset($form['field_room_type_target_id'])) {
  ...set the process and prerender in a copy
}
else {
  ...explicitly define
}